src/zenml/artifacts/artifact_config.py

```diff
@@ -63,32 +63,47 @@ def my_step() -> Annotated[
     """
 
     name: Optional[str] = None
-    version: Optional[Union[str, int]] = None
+    version: Optional[Union[str, int]] = Field(
+        default=None, union_mode="smart"
+    )
     tags: Optional[List[str]] = None
     run_metadata: Optional[Dict[str, MetadataType]] = None
 
     model_name: Optional[str] = None
-    model_version: Optional[Union[ModelStages, str, int]] = None
+    model_version: Optional[Union[ModelStages, str, int]] = Field(
+        default=None, union_mode="smart"
+    )
     is_model_artifact: bool = False
     is_deployment_artifact: bool = False
 
-    @root_validator
-    def _root_validator(cls, values: Dict[str, Any]) -> Dict[str, Any]:
-        model_name = values.get("model_name", None)
-        if model_name and values.get("model_version", None) is None:
-            raise ValueError(
-                f"Creation of new model version from `{cls}` is not allowed. "
-                "Please either keep `model_name` and `model_version` both "
-                "`None` to get the model version from the step context or "
-                "specify both at the same time. You can use `ModelStages.LATEST` "
-                "as `model_version` when latest model version is desired."
-            )
-        return values
+    # TODO: In Pydantic v2, the `model_` is a protected namespaces for all
```

Comment:
This will go away, once we confirm MCP renaming to something else 👍🏼

Reply:
There are many instances of this. I really hope they will go away some day. 😄 

However, in order to keep track of them, I am just leaving the TODOs in the code.